suave/sol/standard_peekers/example.sol

```diff
@@ -44,6 +52,65 @@ contract ExampleEthCallSource {
 
         return abi.encodeWithSelector(this.emptyCallback.selector);
     }
+
+    function findStartIndex(bytes memory data) private pure returns (uint256) {
+        for (uint256 i = 0; i < data.length; i++) {
+            if (data[i] == bytes1(0xff)) {
+                return i;
+            }
+        }
+        return data.length; // Not found
+    }
+
+    event XX(uint256 indexed num, bytes data);
+
+    modifier decodeLogs() {
+        // revert("2");
+
+        bytes memory inputData = msg.data;
+        uint256 magicSequenceIndex = findStartIndex(inputData);
+        require(magicSequenceIndex != inputData.length, "Magic sequence not found");
+
+        // because we have to skip the magic number
+        magicSequenceIndex += 1;
+
+        // Calculate the length of the data to decode
+        uint256 dataLength = inputData.length - magicSequenceIndex;
+
+        // Initialize memory for the data to decode
+        bytes memory dataToDecode = new bytes(dataLength);
+
+        // Copy the data to decode into the memory array
+        for (uint256 i = 0; i < dataLength; i++) {
+            dataToDecode[i] = inputData[magicSequenceIndex + i];
+        }
+
+        emit XX(magicSequenceIndex, dataToDecode);
+
+        // Decode logs from the extracted data
+        Log[] memory logs = abi.decode(dataToDecode, (Log[]));
+
+        /*
+        for (uint256 i = 0; i < logs.length; i++) {
+            emit LogEvent(logs[i].addr);
+        }
+        */
+
+        // Call the function with decoded logs
+        _;
+    }
+
+    function emitLogCallback(uint256 num) public decodeLogs {
```

Comment:
`decodeLogs` is the only modifier the callback needs to emit the logs from the confidential request.